### ISSUE TITLE
helm chart nonRoot nfs

### ISSUE THREAD

**Issue by skillpoint-dev** [bitnami/bitnami-docker-redmine#162]
**Description**

Trying to install redmine via helm chart with persistence via nfs-provisioner. 

**Steps to reproduce the issue:**

1. create values yaml and override container security context with proper UID
2. install with helm chart and values.yaml

**Describe the results you received:**

mkdir: cannot create directory '/opt/bitnami/redmine/tmp/pids': Permission denied

**Describe the results you expected:**

Installation going through like in other charts running as nonRoot e.g. bitnami/ownCloud

**Comment by rafariossaa** [comment]
Hi,
  Could you check the permissions of the attached volume ?

**Comment by skillpoint-dev** [comment]
As mentioned I have nfs-provisioner taking care of persistence so I just assign the storage class defined for it. On the NFS share I have a certain UID that has full access. I set the containerSecurityContext both the UID and GID (just to be sure) of this user. In this way I have multiple instances of e.g bitnami/redis and bitnami/postgresql running...

**Comment by rafariossaa** [comment]
Hi,
  Could you try enabling `volumePermissions`?

**Comment by skillpoint-dev** [comment]
After simple setting the ContainerSecurityContext failed I already tried to enable the "volumePermissions" with, unfortunately, the same result.

**Comment by rafariossaa** [comment]
Hi,
  Could you get into the running pod and run the following ?
```
$ kubectl exec -it myredmine-7b89468cdd-nvq96 bash
root@myredmine-7b89468cdd-nvq96:/# ls -ln /opt/bitnami/redmine/tmp/pids/
total 4
-rw-r--r-- 1 1001 1001 3 Apr 27 16:08 redmine.pid
-rw-r--r-- 1 1001 1001 0 Apr 27 16:08 redmine.pid.lock
```

**Comment by skillpoint-dev** [comment]
Container goes too fast into "Error" and then "CrashLoopBackoff" and I am thrown out of the pod/container if I even make it in there.

**Comment by rafariossaa** [comment]
Hi,
  Could you paste a diff of the original values.yaml and the values.yaml you set ? 
  I would like to try deploying it by myself and try to spot the issue here.

**Comment by skillpoint-dev** [comment]
I deploy a postgresql instance in the same namespace before deploying via bitnami/redmine helm chart

```bash
helm upgrade -i --create-namespace redmine bitnami/redmine -n redmine -f values.yaml
```

```values.yaml
redmineUsername: #MY-USERNAME#
redminePassword: "#MY-PASSWORD#"
containerSecurityContext:
  enabled: true
  runAsUser: #MY-UID#
  runAsGroup: #MY-GID#
podSecurityContext:
  enabled: true
  fsGroup: 1001
volumePermissions:
  enabled: true
persistence:
  storageClass: "MY-STORAGE-CLASS"
databaseType: postgresql
mariadb:
  enabled: false
externalDatabase:
  host: postgresql
  name: redmine
  user: postgres
  password: "MY-DB-PASSWORD"
  port: 5432
service:
  type: ClusterIP
ingress:
  enabled: true
  certManager: true
  annotations:
    kubernetes.io/ingress.class: nginx
    cert-manager.io/cluster-issuer: cloudflare-issuer 
  hostname: #MY-REDMINE-SUBDOMAIN#
  tls: true
```

(values like "MY-..." are obviously replacements of the real values I use...)

Everything starts properly except the redmine pod itself. I see two folders created inside my nfs share by nfs-provisioner one for the postgresql data and one for redmine itself.

I tried all combinations of containerSecurityContext and podSecurityContext. Both on, either off GID provided GID not provided etc. - all resulting in the same error.

As mentioned before I have the same "setup" for bitnami/ownCloud, bitnami/redis, bitnami/postgresql etc. running and working.

**Comment by rafariossaa** [comment]
Hi,
  I was able to reproduce the issue.
  It happens if you use the `containerSecurityPolicy`. 
  I have created an internal task to review a find a solution. 
  Also I would like to point that we are currently migrating all our assets to bash, so this image will be changed in the near future.
  We will come back as soon as we have news.

**Comment by marcosbc** [comment]
Hi, FYI we recently released a new container version, [4.2.1-debian-10-r70](https://github.com/bitnami/bitnami-docker-redmine#421-debian-10-r70). The initialization logic has received a major refactor, and its scripts are now located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-redmine/blob/master/4/debian-10/rootfs/).

Unfortunately we have not made progress for this specific issue, but now it will be a lot easier if you want to contribute non-root support. Note that you may need to change the server from Passenger to a different server, such as Puma or even `rails server`. Any contributions in that direction are welcome and we'll be glad to help with the release.

**Comment by carrodher** [comment]
Hi, unfortunately, if the issue is not solved in recent versions means the task is still on our backlog and there are no plans to work on it in the short/middle term.

In the same way, please, [note the Bitnami team is not actively working on new features/improvements nor providing support through GitHub Issues for this project](https://github.com/bitnami/bitnami-docker-redmine#community-supported-solution) since this asset was marked as a “Community supported solution”.

At this point, the best option to fix any issue or add a new feature is to contribute create a PR; the Bitnami team will be happy to review it and provide feedback.
